### ISSUE TITLE
Search on Rooms is not working

### ISSUE THREAD

**Issue by focafunk** [vector-im/element-web#15120]
![element2](https://user-images.githubusercontent.com/38109047/92487004-1adf9b00-f1c3-11ea-826a-e903ce5310ff.png)
![elem1](https://user-images.githubusercontent.com/38109047/92487012-1c10c800-f1c3-11ea-8336-8168663c9b19.png)
![elem2](https://user-images.githubusercontent.com/38109047/92487089-35197900-f1c3-11ea-898b-ea492ae3c63b.png)

### Description

When you type a search in the search box, always returns _0 results_
It used to work properly until some weeks ago.
### Steps to reproduce

- Type any word existing on any room or selecting _All Rooms_
- No results is returned in search,

It used to work on Riot, and it is the main reason for us to use Element, being able to search in our history of conversations to avoid repeating subjects and keep knowledge
 

Logs being sent: yes/no

### Version information

- **Platform**: web (in-browser) or desktop?
BOTH

For the web app:

- Brave Versión 1.10.93 Chromium: 83.0.4103.106 (Official Build) (64-bit)
- Ubuntu
- https://app.element.io/#/room/!WFMCYJxGKbleFJNsvW:matrix.org

For the desktop app:

-Ubuntu
-Ubuntu 20.04.1 LTS

**Comment by t3chguy** [comment]
The room is encrypted, are you using a client with Seshat enabled?

For Web it is not supported as it says in Settings > Security & Privacy > Message Search
but on the official desktop builds it is. Check that same place to see its indexing progress.

![image](https://user-images.githubusercontent.com/2403652/92487856-b5e96c80-f1e5-11ea-9101-df679ca67262.png)

**Comment by focafunk** [comment]
![elem3](https://user-images.githubusercontent.com/38109047/92488742-54190a80-f1c5-11ea-903e-3375230ca479.png)
This screenshot was taken using Element Desktop, it looks as if it understand I am using the web version
The desktop version is the following one
![elem4](https://user-images.githubusercontent.com/38109047/92488902-8296e580-f1c5-11ea-9c92-ac679b2ec32b.png)

**Comment by focafunk** [comment]
I've installed when it was **riot**, using _sudo apt install_ .Later it updated as Element when the name was changed.
Maybe I should try to re-install it?

**Comment by focafunk** [comment]
I have tried removing the packages _riot-desktop_ and _element-desktop_ and re-installing from scratch following the commands provided in the website, but the same issue is still happening. The versión provided in _About us_ 

**versión de Element: 1.7.5
versión de olm: 3.1.3**

Any ideas? 

These are the commands I've executed

> sudo apt remove riot-desktop
> 
> sudo apt remove element-desktop
> 
> sudo apt install -y wget apt-transport-https
> 
> sudo wget -O /usr/share/keyrings/riot-im-archive-keyring.gpg https://packages.riot.im/debian/riot-im-archive-keyring.gpg
> 
> echo "deb [signed-by=/usr/share/keyrings/riot-im-archive-keyring.gpg] https://packages.riot.im/debian/ default main" | sudo tee /etc/apt/sources.list.d/riot-im.list
> 
> sudo apt update
> 
> sudo apt install element-desktop

**Comment by t3chguy** [comment]
> If you quit Element-Desktop and run it from the Terminal, `element-desktop` are there any warnings or errors?

**Comment by focafunk** [comment]
> Where did you install the desktop version from?
> If you quit Element-Desktop and run it from the Terminal, `element-desktop` are there any warnings or errors?

Sorry, I haven't read the second comment.... yes it seems there is some issue with Buffer

```
rodolfo@RRThinkpad:~$ element-desktop
/home/rodolfo/.config/Element exists: no
/home/rodolfo/.config/Riot exists: yes
Using legacy user data path: /home/rodolfo/.config/Riot
(node:27393) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Starting auto update with base URL: https://packages.riot.im/desktop/update/
Auto update not supported on this platform

```

**Comment by t3chguy** [comment]
Do you still see the same message in Settings > Security & Privacy > Message search?

**Comment by focafunk** [comment]
![image](https://user-images.githubusercontent.com/38109047/92500165-233fd200-f1d3-11ea-8029-478d38e66216.png)
Yes, it still the same

**Comment by jryans** [comment]
Ah it looks like from the logs, there's an error coming from the indexing component:

```
EventIndex: Error initializing the event index {"message":"Error opening the database: IndexError(IOError(IOError { path: None, err: Custom { kind: Other, error: \"Invalid MAC\" } }))"}
```

@poljar, what would you recommend in this case to get indexing working again...? Clearing out any existing versions on disk and restarting Element perhaps?

**Comment by jryans** [comment]
I also filed https://github.com/vector-im/element-web/issues/15168 to improve the diagnostics for this case.

**Comment by poljar** [comment]
> Ah it looks like from the logs, there's an error coming from the indexing component:
> 
> ```
> EventIndex: Error initializing the event index {"message":"Error opening the database: IndexError(IOError(IOError { path: None, err: Custom { kind: Other, error: \"Invalid MAC\" } }))"}
> ```
> 
> @poljar, what would you recommend in this case to get indexing working again...? Clearing out any existing versions on disk and restarting Element perhaps?

I'm afraid that's the only way, either the encryption key got corrupted or the index. Restarting Element shouldn't be needed since the enable/disable cycle works correctly without a restart.

**Comment by jryans** [comment]
@poljar Do you happen to remember which files they are on Linux?

**Comment by poljar** [comment]
The files are in the `EventStore` folder of the electron profile directory, defined over [here](https://github.com/vector-im/element-desktop/blob/develop/src/electron-main.js#L249).

If you are asking about the specific key that is responsible for the index decryption, it will be in the `EventStore/seshat-index.key` file.

Deleting the whole folder is fine, like we already do in [some](https://github.com/vector-im/element-desktop/blob/develop/src/electron-main.js#L528) places.

**Comment by jryans** [comment]
Thanks! @focafunk, could you give the above a try and see if it gets indexing working for you?

**Comment by focafunk** [comment]
Dear @jryans: I've tried deleting all the finles under 

`./confg/Riot/EventoStore ` 

but nothing different happened.
Also I've tryed restarting Element, again with no different results, I cannot search within the rooms yet.

**Comment by focafunk** [comment]
hi guys, any news about this issue?

**Comment by diegofernandess** [comment]
Well, this is curious, because still on version 1.7.15 I had the same issue as  @focafunk . Deleting the directory .config/Element/EventStore solved it.

**Comment by focafunk** [comment]
Hi @diegofernandess  ,unfortunatelly  I do not have such directory. Under Element there is no EventStore as you could see in the screenshot. I will try to delete everything on that directory and try again. Thanks!

![image](https://user-images.githubusercontent.com/38109047/103297155-7d5ce100-49d6-11eb-88fe-69fa6241012f.png)

**Comment by rgpublic** [comment]
For me, the folder was/is still under ~/.config/Riot (the old name of Element). And I had success deleting the EventStore folder to bring the search function back to life.

**Comment by focafunk** [comment]
Finally it worked! 
I noticed what @rgpublic says, I had both folders, Riot and Element
What I did was the following
1 - Quit Element
2 - rm -rf .config/Riot (deleted only Riot,not Element
3 - apt remove element-desktop
4 - apt install element-desktop
5 - rm -rf config/Element
6 - start Element and Sign in

And it finally worked!
Thanks everybody for your precious time and help!

![image](https://user-images.githubusercontent.com/38109047/103300591-f4e23e80-49dd-11eb-8b33-d71a44bf3fc9.png)
